common/djangoapps/third_party_auth/models.py

```diff
@@ -67,8 +67,13 @@ def clean_json(value, of_type):
 
 
 def clean_username(username=''):
-    """ Simple helper method to ensure a username is compatible with our system requirements. """
-    return ('_').join(re.findall(r'[a-zA-Z0-9\-]+', username))[:USERNAME_MAX_LENGTH]
+    """
+    Simple helper method to ensure a username is compatible with our system requirements.
+    """
+    if settings.FEATURES.get("ENABLE_UNICODE_USERNAME"):
+        return ('_').join(re.findall(settings.USERNAME_REGEX_PARTIAL, username))[:USERNAME_MAX_LENGTH]
+    else:
+        return ('_').join(re.findall(r'[a-zA-Z0-9\-]+', username))[:USERNAME_MAX_LENGTH]
```

Comment:
Should there be a test for partial matches of `r'[a-zA-Z0-9\-]+'`?

Reply:
added check for mixed username

Reply:
Thanks. Not a blocker, but I noticed that neither of the `USERNAME_MAX_LENGTH` usages are tested (unless that falls under a different test).